### PR TITLE
review2

### PR DESCRIPTION
**Pull request by hunter-t**

### CODE REVIEW

File: library/src/main/res/drawable/calendar_date_bg_selector.xml [hunk 1]
```diff
@@ -1,6 +1,10 @@
 <?xml version="1.0" encoding="utf-8"?>
 <selector xmlns:android="http://schemas.android.com/apk/res/android" xmlns:app="http://schemas.android.com/apk/res-auto">
 
+    <!--
+        Do we need this state? With this state you can see barely visible selection
+        with NoDateSelectionStrategy, which looks excessive...
+    -->
```
Comment (alexsurinov) [inline library/src/main/res/drawable/calendar_date_bg_selector.xml:7]: Yes, we do. By default `pressed` state should be defined. But you can easily define your custom selector without `pressed` state

File: library/src/main/java/ru/cleverpumpkin/calendar/CalendarDateView.kt [hunk 2]
```diff
@@ -75,13 +75,27 @@ class CalendarDateView @JvmOverloads constructor(
...
     override fun onMeasure(widthMeasureSpec: Int, heightMeasureSpec: Int) {
+        // TODO: No need to call super.onMeasured() and then setMeasuredDimension()
+        // TODO: again, this makes every view being measured twice.
+        // TODO:
+        // TODO: All that you need:
+        // TODO:
+        // TODO: setMeasuredDimension(
+        // TODO:     getDefaultSize(suggestedMinimumWidth, widthMeasureSpec),
+        // TODO:     getDefaultSize(suggestedMinimumWidth, widthMeasureSpec)
+        // TODO: )
```
Comment (alexsurinov) [inline library/src/main/java/ru/cleverpumpkin/calendar/CalendarDateView.kt:98]: Thx, fixed

File: library/src/main/java/ru/cleverpumpkin/calendar/CalendarDateView.kt [hunk 3]
```diff
@@ -75,13 +75,27 @@ class CalendarDateView @JvmOverloads constructor(
     override fun onDraw(canvas: Canvas) {
         textPaint.color = textColor
 
+        // TODO: val xPos = canvas.width / 2.0f <--- don't need to explicitly write ".toFloat()"
         val xPos = (canvas.width / 2).toFloat()
+
+        // TODO: "canvas.height / 2" here makes integer division, which is not accurate enough,
+        // TODO: you're loosing the fractional part, and it possibly may lead to non pixel-perfect
+        // TODO: look of the view. Would be better use Float digit literals in every division.
```
Comment (alexsurinov) [inline library/src/main/java/ru/cleverpumpkin/calendar/CalendarDateView.kt:83]: Ok, fixed

File: library/src/main/java/ru/cleverpumpkin/calendar/CalendarDate.kt [hunk 4]
```diff
@@ -21,6 +21,8 @@ class CalendarDate(date: Date) : Parcelable, Comparable<CalendarDate> {
 
     companion object {
 
+        // TODO: Is there any difficulty with applying @Parcelize annotation above
+        // TODO: the class header instead of writing Parcelable implementation boilerplate code?
```
Comment (alexsurinov) [inline library/src/main/java/ru/cleverpumpkin/calendar/CalendarDate.kt:25]: As I understand, @Parcelize is still experimental feature, so I prefer explicit implementation.  Furthermore, there are not too many Parcelable classes

File: .gitignore [hunk 5]
```diff
@@ -11,6 +11,10 @@ captures/
 
 # Intellij
 *.iml
+
+#
+# Replace all of this with just ".idea/"
+#
```
Comment (alexsurinov) [inline .gitignore:17]: fixed

File: library/src/main/java/ru/cleverpumpkin/calendar/CalendarDate.kt (hunk 4 above)
Comment (hunter-t) [reply library/src/main/java/ru/cleverpumpkin/calendar/CalendarDate.kt:25]: Got it, I think this is a good concern 👍

**Pull request closed without merging**